### DISCUSSION TITLE
file_get_contents(): Failed to open stream: No such file or directory

### DISCUSSION THREAD

**Discussion by Elmatbaisaad** [symfony/symfony#46746] [Q&A]
Hello everyone !
I want to read a jsonfile but its not in the same directory of the code i keep get the  **No such file or directory** .
How can i solve that ?
Thank you !

**Comment by DylanKas** [comment]
We are gonna need more debugging detail to answer, where did you put your file ? How did you use file_get_contents ?

**Reply by Elmatbaisaad** [reply]
thank you @DylanKas  for your answer .
here is my methode to read json file.
![image](https://user-images.githubusercontent.com/101296742/175295651-d854375f-ed2c-4d67-855f-c3009b51d10a.png)
and here is how i use it :
![image](https://user-images.githubusercontent.com/101296742/175295849-a96b0223-9b39-4343-8f9f-ba8034a90d29.png)

**Reply by Elmatbaisaad** [reply]
i tried __Dir__ and __File__ but it doesn't work.

**Reply by stof** [reply]
`dirname('test/AR1.json')` will be `'test'`. I find it totally expected that `file_get_contents('test')` fails.

**Reply by stof** [reply]
and this has nothing to do with Symfony.

**Reply by Elmatbaisaad** [reply]
what should i do to read the file ?

**Reply by stof** [reply]
Pass the path to the file and not a broken one.

**Reply by Elmatbaisaad** [reply]
please @stof i didn't get it the path of my code is /home but the json file  is /home/test how should i call the file in the method ?

**Reply by DylanKas** [reply]
Just use the proper path of your json file without using dirname.

But as stof mentioned, this question is not related to symfony and should not be asked here.
I keep reading some of your questions on this github discussions and I would highly recommend you try some symfony tutorial (and/or PHP as well). You would learn a lot and it will be easier to do whatever you want to later on.

**Reply by Elmatbaisaad** [reply]
i tried but it didn't work 
i willl keep searching thank @DylanKas and @stof  a lot have a nice day
